### ISSUE TITLE
Free all sprites!! (Load free sprites into the game)

### ISSUE THREAD

**Issue by TheCycoONE** [CorsixTH/CorsixTH#545]
_From [Alberth2...@gmail.com](https://code.google.com/u/117559702027655025914/) on April 21, 2013 14:21:15_

Today I spend a day to get the SDL backend to load my new desk into the game. The graphics are bad to say the least, and the code is hacky, but I have a hovering orange desk in my game (if you put it in the right orientation :p )
The code is mostly "proof of concept".

In the "my_sprites" directory is my desk sprite spr2_2355.png, and a .ini file where the section name defines the file to write to and a line like '2355 = "spr2_2355.png", 0+75, 0+66' means that sprite 2355 now comes from "spr2_2355.png", starting at position (0, 0), size (75, 66) ('0+75' is the X interval, '0+66' is the Y interval).
The added make_sprite_files.py program (you need the Python Image Library (PIL) as well, note that the latter implies that you need to use Python 2 afaik) then constructs the file (ie "my_sprites/Data/V-Spr-0.sprites"). It is a very simple format, 2 bytes sprite number, 2 bytes width, 2 bytes height, and width*height bytes, one for each pixel. Sprites are written in increasing sprite number order.

In the C++ code, the sprite is taken from the new file, if the sprite number matches, else it loads from the old data file.
This should take care of any need to distribute modified game data files.

I tried to convince the game to check for existence of the .sprites file, but it wouldn't work, so I resorted to hard-coding the name in the Lua code. Someone with more understanding of Lua should be able to fix this.

I reverse engineered the palette indices to use for my desk. That needs to be documented somewhere so graphics artists can access that information. It also leads to the question of what the project wants here. To me, having so many palettes seems overkill. Perhaps the conversion program to the .sprites file can do something here.

I used Python because it works well for these experiments. For production, a simpler executable may be nice, as not everybody has Python available. The data file format could also use some improvements.

There is the room for making all 4 orientations for all objects now, so you can have a rotating hospital if you like. That should be discussed before starting to make graphics imho.

No doubt I have forgotten some major points, feel free to add/extend :)

**Attachment:** [my_sprites.tar.gz uit.patch orange_desk.png](http://code.google.com/p/corsix-th/issues/detail?id=1593)

_Original issue: http://code.google.com/p/corsix-th/issues/detail?id=1593_

**Comment by Lego3** [comment]
Since it is possible to load sprites into the game now (and animations) I'll close this issue. The next step is to actually get some real new graphics, and to make Lua use them. I suggest that in case we need to track it here we create a new issue.
